app/helpers/sufia/sufia_helper_behavior.rb

```diff
@@ -117,7 +117,7 @@ def search_form_action
 
     def render_visibility_link(document)
       link_to render_visibility_label(document),
-              main_app.edit_curation_concerns_file_set_path(document, anchor: "permissions_display"),
+              main_app.edit_curation_concerns_generic_work_path(document, anchor: "permissions_display"),
```

Comment:
I was thinking about that too, and I looked at the file set visibility badges, and they are not clickable, so they only show the visibility status and does not call the method render_visibility_link. The only way to change the visibility of the FileSet is to go through the Edit option